src/tools/miri/src/shims/unix/sync.rs

```diff
@@ -853,14 +854,23 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                 return interp_ok(());
             }
         };
-        if data.clock == TimeoutClock::RealTime {
-            this.check_no_isolation("`pthread_cond_timedwait` with `CLOCK_REALTIME`")?;
-        }
+
+        let (clock, anchor) = if relative {
+            // `pthread_cond_timedwait_relative_np` always measures time against the
+            // monotonic clock, regardless of the condvar clock.
+            (TimeoutClock::Monotonic, TimeoutAnchor::Relative)
```

Comment:
So the argument named `relative` really is very specific for this macos operation, and also swaps out the clock.  Please give it a name that more accurately reflects what it does (e.g. `macos_relative_np`), and document it in the doc comment for this function.

Reply:
Done, thank you!